svelte-society/sveltesociety.dev
# refactor: Update workspace dependencies and tools

Hi there! I had some ideas for improving the site, but noticed the dev workspace hadn't been updated in a while. This PR modernises some things which will hopefully make it easier to add more exciting features :)

Main changes:
- Switch npm to pnpm
- Update to svelte v4
- Update most other packages
- Switch eslint-plugin-svelte3 to eslint-plugin-svelte
- Switch jest to vitest
- Use latest create-svelte defaults (e.g. app.d.ts, app.html)
- Add svelte-check (not automatically run yet as there were existing errors beyond the scope of this PR)
